platform/security/handler/security-handler-basic/src/main/java/org/codice/ddf/security/handler/basic/AbstractBasicAuthenticationHandler.java

```diff
@@ -0,0 +1,159 @@
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.HttpHeaders;
+
+import org.codice.ddf.security.handler.api.AuthenticationHandler;
+import org.codice.ddf.security.handler.api.BaseAuthenticationToken;
+import org.codice.ddf.security.handler.api.HandlerResult;
+import org.codice.ddf.security.policy.context.ContextPolicy;
+import org.opensaml.xml.util.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractBasicAuthenticationHandler implements AuthenticationHandler {
+
+    public static final String AUTHENTICATION_SCHEME_BASIC = "Basic";
+
+    public static final String SOURCE = "BasicHandler";
+
+    protected static final Logger LOGGER = LoggerFactory
+            .getLogger(AbstractBasicAuthenticationHandler.class);
+
+    @Override
+    public abstract String getAuthenticationType();
+
+    /**
+     * Processes the incoming request to retrieve the username/password tokens. Handles responding
+     * to the client that authentication is needed if they are not present in the request.
+     * Returns the {@link org.codice.ddf.security.handler.api.HandlerResult} for the HTTP Request.
+     *
+     * @param request  http request to obtain attributes from and to pass into any local filter chains required
+     * @param response http response to return http responses or redirects
+     * @param chain    original filter chain (should not be called from your handler)
+     * @param resolve  flag with true implying that credentials should be obtained, false implying return if no credentials are found.
+     * @return
+     */
+    @Override
+    public HandlerResult getNormalizedToken(ServletRequest request, ServletResponse response,
+            FilterChain chain, boolean resolve) {
+
+        String realm = (String) request.getAttribute(ContextPolicy.ACTIVE_REALM);
+        HandlerResult handlerResult = new HandlerResult(HandlerResult.Status.NO_ACTION, null);
+        handlerResult.setSource(realm + "-" + SOURCE);
+
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        String path = httpRequest.getServletPath();
+        LOGGER.debug("Handling request for path {}", path);
+
+        LOGGER.debug("Doing authentication and authorization for path {}", path);
+
+        BaseAuthenticationToken token = extractAuthenticationInfo(httpRequest);
+
+        // we found credentials, attach to result and return with completed status
+        if (token != null) {
+            handlerResult.setToken(token);
+            handlerResult.setStatus(HandlerResult.Status.COMPLETED);
+            return handlerResult;
+        }
+
+        // we didn't find the credentials, see if we are to do anything or not
+        if (resolve) {
+            doAuthPrompt(realm, (HttpServletResponse) response);
```

Comment:
It's small benefit, but you could refactor this and the guts of `handleError` into a common private method.